modules/client/src/main/java/org/apache/ignite/internal/client/sql/ClientAsyncResultSet.java

```diff
@@ -134,11 +136,11 @@ public CompletionStage<? extends AsyncResultSet> fetchNextPage() {
         requireResultSet();
 
         if (closed) {
-            return CompletableFuture.failedFuture(new IgniteClientException("Cursor is closed."));
+            return CompletableFuture.failedFuture(new IgniteException(UNKNOWN_ERR, "Cursor is closed."));
         }
 
         if (!hasMorePages()) {
-            return CompletableFuture.failedFuture(new IgniteClientException("No more pages."));
+            return CompletableFuture.failedFuture(new IgniteException(UNKNOWN_ERR, "No more pages."));
```

Comment:
Do we use `UNKNOWN_ERR` as generic error code?

Reply:
In most cases we should not. I've added TODOs here with a link to [IGNITE-17135 SQL API: Exceptions are different on server and client](https://issues.apache.org/jira/browse/IGNITE-17135).